src/open_feature_client.py

```diff
@@ -0,0 +1,197 @@
+        ).value
+
+    def get_string_details(
+        self,
+        key: str,
+        default_value: str,
+        evaluation_context: typing.Any = None,
+        flag_evaluation_options: typing.Any = None,
+    ) -> FlagEvaluationDetails:
+        return self.evaluate_flag_details(
+            FlagType.STRING,
+            key,
+            default_value,
+            evaluation_context,
+            flag_evaluation_options,
+        )
+
+    def get_number_value(
+        self,
+        key: str,
+        default_value: Number,
+        evaluation_context: typing.Any = None,
+        flag_evaluation_options: typing.Any = None,
+    ) -> Number:
+        return self.evaluate_flag_details(
+            FlagType.NUMBER,
+            key,
+            default_value,
+            evaluation_context,
+            flag_evaluation_options,
+        ).value
+
+    def get_number_details(
+        self,
+        key: str,
+        default_value: Number,
+        evaluation_context: typing.Any = None,
+        flag_evaluation_options: typing.Any = None,
+    ) -> FlagEvaluationDetails:
+        return self.evaluate_flag_details(
+            FlagType.NUMBER,
+            key,
+            default_value,
+            evaluation_context,
+            flag_evaluation_options,
+        )
+
+    def get_object_value(
+        self,
+        key: str,
+        default_value: dict,
+        evaluation_context: typing.Any = None,
+        flag_evaluation_options: typing.Any = None,
+    ) -> dict:
+        return self.evaluate_flag_details(
+            FlagType.OBJECT,
+            key,
+            default_value,
+            evaluation_context,
+            flag_evaluation_options,
+        ).value
+
+    def get_object_details(
+        self,
+        key: str,
+        default_value: dict,
+        evaluation_context: typing.Any = None,
+        flag_evaluation_options: typing.Any = None,
+    ) -> FlagEvaluationDetails:
+        return self.evaluate_flag_details(
+            FlagType.OBJECT,
+            key,
+            default_value,
+            evaluation_context,
+            flag_evaluation_options,
+        )
+
+    def evaluate_flag_details(
+        self,
+        flag_type: FlagType,
+        key: str,
+        default_value: typing.Any,
+        evaluation_context: typing.Any = None,
+        flag_evaluation_options: typing.Any = None,
+    ) -> FlagEvaluationDetails:
+        try:
+            if flag_type is FlagType.BOOLEAN:
+                return self.provider.get_boolean_details(
+                    key,
+                    default_value,
+                    evaluation_context,
+                    flag_evaluation_options,
+                )
+
+            elif flag_type is FlagType.NUMBER:
+                return self.provider.get_number_details(
+                    key,
+                    default_value,
+                    evaluation_context,
+                    flag_evaluation_options,
+                )
```

Comment:
```suggestion
            args = (key, default_value, evaluation_context, flag_evaluation_options)
            if flag_type is FlagType.BOOLEAN:
                return self.provider.get_boolean_details(*args)
            elif flag_type is FlagType.NUMBER:
                return self.provider.get_number_details(*args)
                
# or..
method = TYPE_TO_METHOD_NAME[flag_type]
return getattr(self.provider, method)(key, default_value, evaluation_context, flag_evaluation_options)
```

Possible things to reduce code duplication that I'm sure you probably already know. Not sure if you feel like it's a maintenance headache, but sharing in case you think it's worth it.